packages/jsapi-components/src/spectrum/Picker/Picker.tsx

```diff
@@ -77,11 +79,19 @@ export function Picker({
     // Set viewport to include the selected item so that its data will load and
     // the real `key` will be available to show the selection in the UI.
     function setViewportFromSelectedKey() {
+      let isCanceled = false;
+
       getItemIndexByValue().then(index => {
-        if (index != null) {
-          setViewport(index);
+        if (index == null || isCanceled) {
+          return;
         }
+
+        setViewport(index);
       });
```

Comment:
Should we also be catching here? Otherwise we could have an uncaught promise. Probably just want to log the error anyway I guess.

Reply:
I went ahead and added an explicit catch / log